foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/FoundationCoreJavaAiopsReferencesMigration.java

```diff
@@ -0,0 +1,93 @@
+package com.boozallen.aissemble.upgrade.migration.v1_7_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.baton.BatonException;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.boozallen.aissemble.upgrade.util.FileUtils.replaceLiteralInFile;
+
+public class FoundationCoreJavaAiopsReferencesMigration extends AbstractAissembleMigration {
+
+    public static final Logger logger = LoggerFactory.getLogger(FoundationCoreJavaAiopsReferencesMigration.class);
+    public static List<String> VERSION_AISSEMBLE_PRE_1_7_0 = new ArrayList<>();
+    public static List<String> VERSION_AISSEMBLE_POST_1_7_0 = new ArrayList<>();
+
+    /**
+     * Function to add packages to the VERSION_AISSEMBLE_PRE_1_7_0 & VERSION_AISSEMBLE_POST_1_7_0 list.
+     * @param n/a.
+     * @return n/a.
+     */
+    public static void addPackagesToList(){
+        VERSION_AISSEMBLE_PRE_1_7_0.add("com.boozallen.aiops.core.cdi");
+        VERSION_AISSEMBLE_PRE_1_7_0.add("com.boozallen.aiops.core.metadata");
+        VERSION_AISSEMBLE_PRE_1_7_0.add("com.boozallen.aiops.core.filestore");
+
+        VERSION_AISSEMBLE_POST_1_7_0.add("com.boozallen.aissemble.core.cdi");
+        VERSION_AISSEMBLE_POST_1_7_0.add("com.boozallen.aissemble.core.metadata");
+        VERSION_AISSEMBLE_POST_1_7_0.add("com.boozallen.aissemble.core.filestore");
+    }
+
+     /**
+     * Function to check whether the migration is necessary.
+     * @param file file to check
+     * @return shouldExecute - whether the migration is necessary.
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+        addPackagesToList();
+
+        try (BufferedReader foundationCoreJavaConfig = new BufferedReader((new FileReader(file)))) {
+
+            String line;
+            while((line = foundationCoreJavaConfig.readLine()) !=null && !shouldExecute) {
+                for (int i=0; i < VERSION_AISSEMBLE_PRE_1_7_0.size(); i++) {
```

Comment:
A: Should be pretty easy and a bit more efficient to use the concatenate the list of literals into a regex `Pattern`.
```java
String regex = oldPackageNames.stream()
   .map(Pattern::quote)
   .collect(Collectors.joining("|"))
Pattern search = Pattern.compile(regex);
```

At that point you can probably reuse some of our util logic.